ada/ast.py

```diff
@@ -13930,50 +13930,45 @@ def env_elements_baseid():
             )
         ))
 
-    @langkit_property(return_type=Equation)
-    def denotes_the_property_function(subp_decl=T.SubpDecl.entity):
+    @langkit_property(return_type=Bool)
+    def denotes_the_property_function(subp_spec=T.SubpSpec.entity):
         # Return true whether this node can refer to a property function
-        # detoned by `subp_decl`. (see RM 7.3.4 about stable properties of a
+        # detoned by `subp_spec`. (see RM 7.3.4 about stable properties of a
         # type). This equation has to be called in the scope of the
         # `Stable_Properties` aspect name resolution.
 
-        primitive_types = Var(subp_decl.subp_spec.primitive_subp_types())
+        primitive_types = Var(subp_spec.primitive_subp_types())
 
         # ``subp_decl`` is a property function of this node if it comes from a
         # `Stable_Properties` AspectAssoc and:
-        return If(
-            And(
-                # It only has one single parameter (mode in but not checked
-                # here).
-                subp_decl.subp_spec.params.length == 1,
-
-                # It matches the type for which the Stable_Properties is
-                # defined. There are two cases:
-                Entity.parent_basic_decl.match(
-                    # Either the Stable_Propeties aspect is defined within a
-                    # TypeDecl.
-                    lambda td=T.TypeDecl: Not(
-                            primitive_types.find(
-                                lambda t:
-                                t.name.matches(td.name.node)
-                            ).is_null
-                        ),
-                    # Or within a SubpDecl
-                    lambda sd=T.SubpDecl: Not(
-                            sd.subp_spec.primitive_subp_types().filter(
-                                lambda t1: Not(
-                                    primitive_types.find(
-                                        lambda t2:
-                                        t2.name.matches(t1.name.node)
-                                    ).is_null
-                                )
-                            ).is_null
-                        ),
-                    lambda _: False
-                )
-            ),
-            LogicTrue(),
-            LogicFalse()
+        return And(
+            # It only has one single parameter (mode in but not checked here)
+            subp_spec.params.length == 1,
+
+            # It matches the type for which the Stable_Properties is defined.
+            # There are two cases:
+            Entity.parent_basic_decl.match(
+                # Either the Stable_Propeties aspect is defined within a
+                # TypeDecl.
+                lambda td=T.TypeDecl: Not(
+                        primitive_types.find(
+                            lambda t:
+                            t.name.matches(td.name.node)
```

Comment:
Why a name comparison? Could you not compare `t` and `td` directly? (same below)

Reply:
I was just confused I guess, thanks for the suggestion!